velociraptor/observations/objects.py

```diff
@@ -531,6 +532,12 @@ def plot_on_axes(self, axes: Axes, errorbar_kwargs: Union[dict, None] = None):
 
         errorbar_kwargs: dict
             Optional keyword arguments to pass to plt.errorbar.
+
+        line_idx: int
```

Comment:
I don't like leaving users to specify which index... perhaps we can have them give a redshift and the closest data is returned?

Reply:
In this version, the user never gets to choose the index. Instead, it is done automatically via minimising **abs**(observational data z - snapshot z)

Reply:
Actually, yes, indeed! There is a much better way of doing this.